compiler/rustc_parse/src/lib.rs

```diff
@@ -257,7 +257,13 @@ pub fn nt_to_tokenstream(
     let tokens = match *nt {
         Nonterminal::NtItem(ref item) => prepend_attrs(sess, &item.attrs, nt, item.tokens.as_ref()),
         Nonterminal::NtBlock(ref block) => convert_tokens(block.tokens.as_ref()),
-        Nonterminal::NtStmt(ref stmt) => prepend_attrs(sess, stmt.attrs(), nt, stmt.tokens()),
+        Nonterminal::NtStmt(ref stmt) => {
+            if let ast::StmtKind::Empty = stmt.kind {
+                Some(tokenstream::TokenTree::token(token::Semi, stmt.span).into())
```

Comment:
As the example in https://github.com/rust-lang/rust/pull/80789#issuecomment-756447607 shows, empty statements can have attributes too.
So `prepend_attrs` also needs to be used here in some form.

Reply:
Fixed